rclpy/src/rclpy/action_client.hpp

```diff
@@ -0,0 +1,237 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY__ACTION_CLIENT_HPP_
+#define RCLPY__ACTION_CLIENT_HPP_
+
+#include <pybind11/pybind11.h>
+
+#include <rcl_action/rcl_action.h>
+
+#include <memory>
+
+#include "destroyable.hpp"
+#include "handle.hpp"
+
+namespace py = pybind11;
+
+namespace rclpy
+{
+class ActionClient : public Destroyable, public std::enable_shared_from_this<ActionClient>
+{
+public:
+  /// Create an action client.
+  /**
+   * This function will create an action client for the given action name.
```

Comment:
@ahcorde nit: I wonder if this should be rephrased into (and put as) class level documentation.